File: test_dir/python/src/dicts/exceptions.py
from test_dir.python.pypp.dict_get import dg


def dict_exceptions_fn():
    print("DICT EXCEPTIONS RESULTS:")
    a: dict[int, int] = {0: 1, 1: 2}
    try:
        a.pop(-1)
    except KeyError as e:
        print("key error: " + str(e))
    try:
        dg(a, -1)
    except KeyError as e:
        print("key error: " + str(e))

File: test_dir/cpp/src/dicts/exceptions.cpp
#include "dicts\exceptions.h"
#include "exceptions/stdexcept.h"
#include "py_dict.h"
#include "py_str.h"
#include "pypp_util/print.h"
#include "pypp_util/to_py_str.h"
#include <string>

void dict_exceptions_fn() {
    print(PyStr("DICT EXCEPTIONS RESULTS:"));
    PyDict<int, int> a({{0, 1}, {1, 2}});
    try {
        a.pop(-1);
    } catch (const PyppKeyError &pypp_e) {
        std::string e = pypp_e.what();
        print(PyStr("key error: ") + to_pystr(e));
    }
    try {
        a.dg(-1);
    } catch (const PyppKeyError &pypp_e) {
        std::string e = pypp_e.what();
        print(PyStr("key error: ") + to_pystr(e));
    }
}
